File: gtsam-rs/sys/src/geometry/cal3_s2.cc
#include "cal3_s2.h"

namespace gtsam
{

    std::shared_ptr<Cal3_S2> default_cal3_s2() {
        return std::make_shared<Cal3_S2>();
    }

    std::shared_ptr<Cal3_S2> new_cal3_s2(double fx, double fy, double s, double u0, double v0)
    {
        return std::make_shared<Cal3_S2>(fx, fy, s, u0, v0);
    }

} // namespace gtsam

File: gtsam-rs/src/geometry/cal3_s2.rs
pub struct Cal3S2 {
    pub(crate) inner: cxx::SharedPtr<sys::Cal3_S2>,
}

impl Cal3S2 {
    pub fn default() -> Self {
        Self {
            inner: sys::default_cal3_s2(),
        }
    }

    pub fn new(fx: f64, fy: f64, s: f64, u0: f64, v0: f64) -> Self {
        Self {
            inner: sys::new_cal3_s2(fx, fy, s, u0, v0),
        }
    }
}